pycargoebuild/__main__.py

```diff
@@ -71,13 +71,21 @@ def main(prog_name: str, *argv: str) -> int:
     load_license_mapping(args.license_mapping)
     args.license_mapping.close()
 
+    def locate_workspace_root(directory: Path) -> Path:
+        for current_dir in (directory.resolve() / "Cargo.lock" ).parents:
+            path = current_dir / "Cargo.lock"
+            if path.exists():
+                return current_dir
+        raise FileNotFoundError
```

Comment:
Could you `open()` them and catch `FileNotExistsError` instead? And then reraise the last exception if no directory contains it.